docs/core/whats-new/dotnet-9/libraries.md

```diff
@@ -565,40 +569,66 @@ The <xref:System.Json.JsonObject> type now exposes ordered dictionary&ndash;like
 
 :::code language="csharp" source="../snippets/dotnet-9/csharp/Serialization.cs" id="PropertyOrder":::
 
-### Additional contract metadata APIs
+### Customizing enum member names
 
-The JSON contract API now exposes additional metadata including constructor metadata information and improved attribute provider support for the case of the source generator.
-
-The new APIs have the following shape:
+The new <xref:System.Text.Json.Serialization.JsonStringEnumMemberNameAttribute?displayProperty=nameWithType> attribute can be used to customize the names of individual enum members for types that are serialized as strings:
 
 ```csharp
-namespace System.Text.Json.Serialization.Metadata;
+JsonSerializer.Serialize(MyEnum.Value1 | MyEnum.Value2); // "Value1, Custom enum value"
 
-public partial class JsonTypeInfo
+[Flags, JsonConverter(typeof(JsonStringEnumConverter))]
+enum MyEnum
 {
-    // Typically the ConstructorInfo of the active deserialization constructor.
-    public ICustomAttributeProvider? ConstructorAttributeProvider { get; }
+    Value1 = 1,
+    [JsonStringEnumMemberName("Custom enum value")]
+    Value2 = 2,
 }
+```
 
-public partial class JsonPropertyInfo
-{
-    public Type DeclaringType { get; }
-    // Typically the FieldInfo or PropertyInfo of the property.
-    public ICustomAttributeProvider? AttributeProvider { get; set; }
-    // The constructor parameter that has been associated with the current property.
-    public JsonParameterInfo? AssociatedParameter { get; }
-}
+### Streaming multiple JSON documents
+
+<xref:System.Text.Json.Utf8JsonReader?displayProperty=nameWithType> now supports reading multiple, whitespace-separated JSON documents from a single buffer or stream. By default, it will throw an exception if it detects any non-whitespace characters that are trailing the first top-level document. This behavior can be changed using the <xref:System.Text.Json.JsonReaderOptions.AllowMultipleValues> flag:
```

Comment:
Does this relate to JsonLines?

https://jsonlines.org/

Reply:
Essentially, although this particular implementation is more forgiving allowing arbitrary whitespace separation between values.